test/test_unit/validation-errors-converter.test.js

```diff
@@ -3,163 +3,226 @@ const SchemaValidator = require('../../src/schema-validator')
 
 describe('validation-error-converter test', () => {
 
-    const schema = {
-        $id: 'http://example.com/schemas/schema.json',
-        type: 'object',
-        additionalProperties: false,
-        properties: {
-            typeError: {
-                type: 'string'
-            },
-            additionalPropertyError: {
-                type: 'object',
-                additionalProperties: false,
-                properties: {
-                    a: {
-                        type: 'string'
-                    }
-                }
-            },
-            requiredError: {
-                type: 'object',
-                required: ['a'],
-                properties: {
-                    a: {
-                        type: 'string'
-                    }
-                }
-            },
-            constError: {
-                type: 'string',
-                const: 'dummy'
-            },
-            enumError: {
-                type: 'string',
-                enum: [
-                    '1',
-                    '2',
-                    '3'
-                ]
-            },
-            propEnumError: {
-                type: 'object',
-                propertyNames: {
-                    enum: [
-                        'a',
-                        'b',
-                        'c'
-                    ]
+    it('type', () => {
+        const schema = {
+            $id: 'http://example.com/schemas/schema.json',
+            type: 'object',
+            additionalProperties: false,
+            properties: {
+                typeError: {
+                    type: 'string'
                 }
-            },
-            formatUrlError: {
-                type: 'string',
-                format: 'httpUrl'
-            },
-            formatError: {
-                type: 'string',
-                format: 'email'
             }
         }
-    }
-    const schemaValidator = new SchemaValidator(schema.$id, [schema])
+        const schemaValidator = new SchemaValidator(schema.$id, [schema])
 
-    it('type', () => {
         const data = {
             typeError: {}
         }
 
-        const result = schemaValidator.validate(data)
-        expect(result).to.include({
+        const result = schemaValidator.validate.bind(schemaValidator, data)
+        expect(result).to.throw().to.include({
             field: '.typeError',
             message: "Property 'typeError' must be string.",
             rejected_value: data.typeError
         })
     })
 
     it('additionalProperties', () => {
+        const schema = {
```

Comment:
I believe this error can be validated directly in the root:
```
field: '.d'
message: 'd is not supported'
```

Of course, you better naming. ;)